### PR TITLE
Add ismulticast(::IPv4) function

### PR DESCRIPTION
**Pull request by klf-gs**
Add a function to check if an IPv4 address is within the IANA multicast address range.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "NetworkProtocols"
 uuid = "cc8e9863-198f-4c11-8208-8a22403099d5"
 authors = ["Christian Rorvik <christian.rorvik@gmail.com>"]
-version = "0.2.5"
+version = "0.3.5"
```
Comment (ancapdev) [inline Project.toml]: We haven't done a 1.0 release here yet, so bumping the first non-zero is for breaking changes, patch/feature changes are the same (last component)
```suggestion
version = "0.2.6"
```

File: test/ip_tests.jl [hunk 2]
```diff
@@ -16,4 +16,13 @@
     @test length(ipp.payload) == 48
 end
 
+@testset "ismulticast" begin
+    ep = decode_ethernet(multicast_packet)
+    ipp = decode_ipv4(ep.payload)
+    @test ismulticast(ipp.header.dst_ip) == true
```
Comment (ancapdev) [inline test/ip_tests.jl]:
```suggestion
    @test ismulticast(ipp.header.dst_ip)
```

File: test/ip_tests.jl [hunk 3]
```diff
@@ -16,4 +16,13 @@
...
+
+    epdns = decode_ethernet(dns_packet)
+    ippdns = decode_ipv4(epdns.payload)
+    @test ismulticast(ippdns.header.dst_ip) == false
```
Comment (ancapdev) [inline test/ip_tests.jl]:
```suggestion
    @test !ismulticast(ippdns.header.dst_ip)
```

File: test/ip_tests.jl [hunk 4]
```diff
@@ -16,4 +16,13 @@
     @test length(ipp.payload) == 48
 end
 
+@testset "ismulticast" begin
```
Comment (ancapdev) [inline test/ip_tests.jl:19]: Why do we construct a whole packet to test a function that operators on IP?

Comment (klf-gs) [reply test/ip_tests.jl:19]: True, this is wasteful of cognitive resources. I'll change this to a simpler test.

Comment (klf-gs) [reply test/ip_tests.jl:19]: Updated with new tests now.

File: src/ip.jl [hunk 5]
```diff
@@ -55,3 +55,7 @@ function decode_ipv4(data::DenseVector{UInt8})
             p + h.header_length,
             (Int(h.total_length - h.header_length),)))
 end
+
+function ismulticast(ip::IPAddr)
```
Comment (green-nsk) [inline src/ip.jl]: comparison will fail for IPv6 addresses
```suggestion
function ismulticast(ip::IPv4)
```

**Pull request merged**